akka-http/src/main/scala/akka/http/javadsl/server/Marshaller.scala

```diff
@@ -4,8 +4,108 @@
 
 package akka.http.javadsl.server
 
-/**
- * A marker trait for a marshaller that converts a value of type [[T]] to an
- * HttpResponse.
- */
-trait Marshaller[T]
\ No newline at end of file
+import java.util.function
+import akka.http.scaladsl.marshalling
+import scala.concurrent.ExecutionContext
+import akka.http.javadsl.model.ContentType
+import akka.http.javadsl.model.MediaType
+import akka.http.scaladsl
+import akka.http.javadsl.model.HttpEntity
+import akka.http.scaladsl.marshalling._
+import akka.http.javadsl.model.HttpResponse
+import akka.http.javadsl.model.HttpRequest
+import akka.http.javadsl.model.RequestEntity
+import akka.util.ByteString
+import akka.http.scaladsl.model.FormData
+import akka.http.javadsl.model.StatusCode
+import akka.http.javadsl.model.HttpHeader
+import scala.collection.JavaConverters._
+import JavaScalaTypeEquivalence._
+
+object Marshaller {
+  def wrap[A,B](scalaMarshaller: marshalling.Marshaller[A,B]) = Marshaller()(scalaMarshaller)
```

Comment:
Sure, those names can be aligned. Speaking of which, the existing conversions also say `asJava` as well as `toJava` (java8-compat, collection convertors I believe).